src/clr-angular/utils/a11y/aria-live.service.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { DOCUMENT } from '@angular/common';
+import { Inject, Injectable, NgZone, OnDestroy } from '@angular/core';
+import { uniqueIdFactory } from '../id-generator/id-generator.service';
+
+// export type AriaLivePoliteness = 'off' | 'polite' | 'assertive';
+export enum AriaLivePoliteness {
+  off = 'off',
+  polite = 'polite',
+  assertive = 'assertive',
+}
+
+@Injectable()
+export class AriaLiveService implements OnDestroy {
+  private _TICK: number = 100;
+  private _ariaLiveElement: HTMLElement;
+  private _document: Document;
+  private _previousTimeout?: number;
+
+  private _id: string = `clr-aria-live-element-${uniqueIdFactory()}`;
+
+  constructor(private _ngZone: NgZone, @Inject(DOCUMENT) _document: any) {
+    this._document = _document;
+    this._ariaLiveElement = this._createLiveElement();
+  }
+
+  private _createLiveElement(): HTMLElement {
+    const ariaLiveElement = this._document.createElement('div');
+
+    ariaLiveElement.setAttribute('id', this.getId());
+    // Use clarity screen reader class to hide the dom element
+    // and fix the scrollbar shake
+    ariaLiveElement.classList.add('clr-sr-only');
+
+    ariaLiveElement.setAttribute('aria-atomic', 'true');
+    ariaLiveElement.setAttribute('aria-live', AriaLivePoliteness.polite);
+
+    this._document.body.appendChild(ariaLiveElement);
+
+    return ariaLiveElement;
+  }
+
+  public announce(message: string, politeness: AriaLivePoliteness = AriaLivePoliteness.polite): Promise<void> {
+    this.clear();
+    clearTimeout(this._previousTimeout);
+
+    this._ariaLiveElement.setAttribute('aria-live', politeness);
+
+    // This 100ms timeout is necessary for some browser + screen-reader combinations:
+    // - Both JAWS and NVDA over IE11 will not announce anything without a non-zero timeout.
+    // - With Chrome and IE11 with NVDA or JAWS, a repeated (identical) message won't be read a
+    //   second time without clearing and then using a non-zero delay.
+    // (using JAWS 17 at time of this writing).
+    return this._ngZone.runOutsideAngular(() => {
+      return new Promise(resolve => {
+        clearTimeout(this._previousTimeout);
+        /**
+         * Cause I didn't want to use `window.setTimeout` so TypeScript will stop throwing
+         * this error `Type 'Timer' is not assignable to type 'number'` , I had to use `<any>` before
+         * setTimeout()
+         *
+         * More on the topic here: https://github.com/TypeStrong/atom-typescript/issues/1053
+         */
+        this._previousTimeout = <any>setTimeout(() => {
```

Comment:
I've encountered this problem before. I think we are better off just having `any` type on the `_previousTimeout` property.

Reply:
I prefer to keep it like that. I think that `_previousTimeout` is the property that I want to know its type. `setTimeout` is just buggy and there is a mismatch between javascript and nodejs types here. The other solution, if I don't want to annotate them with strange types, is to use the `window.setTimeout` but will have to wrap them inside another platform check.

So my point here is, prefer to know my custom property type than to know the type of native method that won't change the next 100 years ;)

Reply:
> <any>setTimeout(() => {

you are type-casting whatever the type of setTimeout to `<any>`. So there is no point in assigning `number` type to the property. If we want to be more strict on the type, 
we should do:
```
private previousTimeout: ReturnType<typeof setTimeout>;
```

Reply:
Ok, let's do it like that :)